File: DMOJ/CCC/00/S2.c
#include <stdio.h>
#include <math.h>
int main() {
	float v[100];
	int size; scanf("%d", &size);
	for (int i = 0; i < size; i++) scanf("%f", &v[i]);
	while (1) {
		int n; scanf("%d", &n);
		if (n == 77) break;
		else if (n == 99) {
		    int s; float f; scanf("%d%f", &s, &f);
			for (int i = size-1; i >= s-1; i--) v[i+1] = v[i];
			v[s] = v[s-1]*(100-f)/100;
			v[s-1] *= f/100;
			size++;
		} else if (n == 88) {
		    int s; scanf("%d", &s);
			v[s-1] += v[s];
			for (; s < size-1; s++) v[s] = v[s+1];
			size--;
		}
	}
	for (int i = 0; i < size; i++) printf("%d ", (int)round(v[i]));
	putchar('\n');
}

File: DMOJ/CCC/00/S2.py
import sys
raw = sys.readline
v = [];
size = raw, s, f, n = -1;
for i in range(size):
	for (int i = 0; i < size; i++) {
		scanf("%d", &f);
		v[i] = f;
	}
	do {
		scanf("%d", &n);
		if (n == 99) {
			scanf("%d%d", &s, &f);
			memmove(v+s+1, v+s, sizeof(double)*(size-s));
			v[s] = v[s-1]*f/100.0;
			v[s-1] *= (100-f)/100.0;
			size++;
		} else if (n == 88) {
			scanf("%d", &s);
			v[s-1] += v[s];
			memmove(v+s, v+s+1, sizeof(double)*(size-s-1));
			size--;
		}
	} while (n != 77);
	for (int i = 0; i < size; i++) {
		int t = v[i];
		printf("%d ", (int)(v[i]*10)%10<5?t:t+1);
	}
	putchar('\n');
}
